File: Sales by Match.c
/* accepted in first attempt */

int sockMerchant(int n, int ar_count, int *ar)
{
    int socksCounter[101];
    int i, pairs = 0;

    for (i = 0; i < 101; i++)
        socksCounter[i] = 0;

    for (i = 0; i < n; i++)
    {
        socksCounter[ar[i]]++;

        if (socksCounter[ar[i]] != 0 && socksCounter[ar[i]] % 2 == 0)
            pairs++;
    }

    return pairs;
}

File: Sales by Match.py
# accepted in first attempt
# Time Complexity: O(n)


def sockMerchant(n, ar):
    socksCounter = [0] * 101
    pairs = 0

    for number in ar:
        socksCounter[number] += 1

        if socksCounter[number] != 0 and socksCounter[number] % 2 == 0:
            pairs += 1

    return pairs
